cybereason_connector.py

```diff
@@ -600,6 +705,251 @@ def _get_malop_sensor_ids(self, malop_id, action_result):
+            err = self._get_error_message_from_exception(e)
+            return action_result.set_status(phantom.APP_ERROR, "Error occurred. {}".format(err))
+
+        return action_result.set_status(phantom.APP_SUCCESS, "Successfully requested for sensor upgrade")
+
+    def _handle_restart_sensor(self, param):
+        self.save_progress("In _handle_restart_sensor function")
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+
+        # Add an action result object to self (BaseConnector) to represent the action for this param
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        # Get the parameters
+        pylum_id = self._get_string_param(param['pylumid'])
+
+        try:
+            # Create a session to call the rest APIs
+            cr_session = CybereasonSession(self).get_session()
+
+            url = "{0}/rest/sensors/action/restart".format(self._base_url)
+            self.save_progress(url)
+
+            pylum_ids = []
+            if "," in pylum_id:
+                filter_arr = pylum_id.strip().split(",")
+                filter_arr = [each_id.strip() for each_id in filter_arr]
+                pylum_ids.extend(filter_arr)
+            else:
+                pylum_ids.append(pylum_id)
+            query = json.dumps({
+                "filters": [
+                    {
+                        "fieldName": "pylumId",
+                        "operator": "ContainsIgnoreCase",
+                        "values": pylum_ids
+                    }
+                ]
+            })
+
+            res = cr_session.post(url, data=query, headers=self._headers)
+            if res.status_code == 204:
+                return action_result.set_status(
+                    phantom.APP_ERROR,
+                    "Status Code:204. The sensor names are incorrect or the filters are not valid"
+                )
+            if res.status_code < 200 or res.status_code >= 399:
+                self._process_response(res, action_result)
+                return action_result.get_status()
+
+            json_res = res.json()
+            self.save_progress("Sensors Restart Requested")
+            action_result.update_summary(json_res)
+            # Data will typically be the raw JSON if we need to use it in a playbook
+            action_result.add_data(json_res)
+
+        except Exception as e:
+            err = self._get_error_message_from_exception(e)
+            return action_result.set_status(phantom.APP_ERROR, "Error occurred. {}".format(err))
+
+        return action_result.set_status(phantom.APP_SUCCESS, "Successfully requested for sensor restart")
+
+    def _get_machine_name_by_machine_ip(self, machine_ip, action_result):
+        machine_names = []
+        try:
+            cr_session = CybereasonSession(self).get_session()
+
+            url = "{0}/rest/sensors/query".format(self._base_url)
+            self.save_progress(url)
```

Comment:
suggestion: Should be add some additional text here to show what the url is about?

Reply:
Added